ambry-router/src/main/java/com.github.ambry.router/PutOperation.java

```diff
@@ -696,9 +696,16 @@ private void cleanupExpiredInFlightRequests() {
           correlationIdToChunkPutRequestInfo.entrySet().iterator();
       while (inFlightRequestsIterator.hasNext()) {
         Map.Entry<Integer, ChunkPutRequestInfo> entry = inFlightRequestsIterator.next();
-        if (time.milliseconds() - entry.getValue().startTimeMs > routerConfig.routerRequestTimeoutMs) {
+        if (time.milliseconds() - entry.getValue().getStartTimeMs() > routerConfig.routerRequestTimeoutMs) {
           operationTracker.onResponse(entry.getValue().replicaId, false);
           chunkException = new RouterException("Timed out waiting for responses", RouterErrorCode.OperationTimedOut);
+          ReplicaId replica = entry.getValue().replicaId;
+          DataNodeId dataNodeId = replica.getDataNodeId();
+          NonBlockingRouterMetrics.NodeLevelMetrics dataNodeBasedMetrics =
+              routerMetrics.getDataNodeBasedMetrics(dataNodeId);
+          dataNodeBasedMetrics.putRequestTimeoutCount.inc();
```

Comment:
agree with you. The old frontend metrics reported ~800 metrics, of which many are for node-level metrics. We should keep per-node metrics number small, as this number will be multiplied by the number of nodes.